### PR TITLE
ci: Add a new workflow to check and format shell scripts in `util` dir

### PR DESCRIPTION
**Pull request by starccy**
Hi,
I've added a new GitHub action workflow to resolve https://github.com/uutils/coreutils/issues/4752. It utilizes `shellcheck` and `shfmt` tools to perform checks and formatting on our shell scripts during a push to the main branch or the creation of a PR.

As it stands, I would appreciate some input on the following aspects that may require further discussion or decision: 

1. Unification of Shell Interpreters: It appears that some of our scripts utilize #!/bin/sh shebang, yet contain certain statements that do not adhere to the POSIX sh standard, as pointed out by shellcheck. I have attached an example of such an occurrence in the CI output from my forked repo here:  [ShellScript/Check](https://github.com/starccy/coreutils/actions/runs/5539084094/jobs/10109672892). As a temporary measure, I've configured `shellcheck` to scrutinize all scripts against the `bash` rules. Would this be acceptable or should we strictly adhere to the $shebang rules?
2. Inclusion of Specific `shellcheck` Rules: Currently, I've left the `shellcheck` rules to their default settings, which still yield a number of warnings: [ShellScript/Check](https://github.com/starccy/coreutils/actions/runs/5539511763/jobs/10110527777). My plan is to address these in an additional commit, but before proceeding, it would be beneficial to determine which rules we should strictly enforce.
3. Auto Commit Post `shfmt` Execution: Once `shfmt` completes successfully, an automatic commit will be added to the main branch. I was wondering if this practice aligns with our workflow expectations.

### CODE REVIEW

File: .github/workflows/CheckScripts.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        run: |
+          # show differs first for every files that need to be formatted
+          # fmt options: bash syntax, 4 spaces indent, indent for switch-case
+          echo "## show the differences between formatted and original scripts..."
+          find ${{ env.SCRIPT_DIR }} -name "*.sh" -print0 | xargs -0 shfmt -ln=bash -i 4 -ci -d || true
+          # perform a shell format
+          echo "## perform a shell format..."
+          # ignore the error code because `-d` will always return false when the file has difference
+          find ${{ env.SCRIPT_DIR }} -name "*.sh" -print0 | xargs -0 shfmt -ln=bash -i 4 -ci -w
+      - name: Commit any changes
```
Comment (sylvestre) [inline .github/workflows/CheckScripts.yml:67]: please replace it by a failure, not an autofix :)

**Comment by sylvestre** [comment]
it didn't run. I guess it is because a new testsuite.
Let's see if it works if I merge it once you fixed the comment

**Comment by tertsdiepraam** [comment]
Yes, I think that's due to that security feature of github where it won't tun unmerged workflows.

File: .github/workflows/CheckScripts.yml (hunk 1 above)
Comment (starccy) [reply .github/workflows/CheckScripts.yml:67]: Got it, I'll remove the autofix code.

**Comment by starccy** [comment]
I've removed any write operations. now the job will fail if the code does not meet the formatting specifications.

**Comment by sylvestre** [comment]
Let's try
Thanks

**Comment by starccy** [comment]
And as I've mentioned above, the job will be triggered if we modify the shell script under the `util` dir. But in the current situation, the `shellcheck`/`shfmt` will complain about some scripts. Should I add an additional commit to fix them?

**Comment by sylvestre** [comment]
yeah, please :)

**Pull request merged**